src/transformers/modeling_utils.py

```diff
@@ -1401,8 +1402,16 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     model = cls(config, *model_args, **model_kwargs)
         else:
             with no_init_weights(_enable=_fast_init):
+
+                restore_gradient_checkpointing = False
+                if hasattr(config, "gradient_checkpointing") and config.gradient_checkpointing:
+                    restore_gradient_checkpointing = True
+
                 model = cls(config, *model_args, **model_kwargs)
 
+                if restore_gradient_checkpointing:
+                    config.gradient_checkpointing = True
+
```

Comment:
I don't think this is necessary, we can just keep the `restore_gradient_checkpointing` bool